registry/logging.go

```diff
@@ -0,0 +1,118 @@
+package registry
+
+import (
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	jsoniter "github.com/json-iterator/go"
+)
+
+var json = jsoniter.ConfigCompatibleWithStandardLibrary
```

Comment:
Perhaps we should use a different name for this, as it's very easy to overlook, and other code may also be using `json` (and switch to the other implementation without realising)

Reply:
I think this is fine -- even the name suggests it's compatible with stdlib 😄